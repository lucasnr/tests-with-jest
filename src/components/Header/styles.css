.header {
	background-color: #d44;
	color: #fff;
	display: flex;
	justify-content: space-between;
	padding: 1rem;
}

.header .header__nav ul {
	display: flex;
}

.header .header__nav ul a {
	padding: 0.25rem 0.75rem;
}
