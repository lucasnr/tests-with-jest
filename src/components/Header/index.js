import React from 'react';

import './styles.css';

function Header() {
	return (
		<header className="header">
			<a href="/" className="header__brand">
				Nav
			</a>
			<nav className="header__nav">
				<ul>
					<li>
						<a href="/">Home</a>
					</li>
					<li>
						<a href="/">Profile</a>
					</li>
					<li>
						<a href="/">Clients</a>
					</li>
				</ul>
			</nav>
		</header>
	);
}

export default Header;
