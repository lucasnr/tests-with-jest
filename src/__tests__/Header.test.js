import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

describe('Header', () => {
	it('renders without crashing and matches snapshot', () => {
		const wrapper = shallow(<Header />);
		expect(wrapper).toMatchSnapshot();
	});
});
