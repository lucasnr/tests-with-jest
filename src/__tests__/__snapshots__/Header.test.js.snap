// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing and matches snapshot 1`] = `
<header
  className="header"
>
  <a
    className="header__brand"
    href="/"
  >
    Nav
  </a>
  <nav
    className="header__nav"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Profile
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Clients
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
